src/jit/compiler.h

```diff
@@ -5710,6 +5711,7 @@ class Compiler
                 __int64 lconVal;
                 double  dconVal;
                 Range   u2;
+                bool    edge;
```

Comment:
Would using an enum instead of a bool make it more clear? Say:
```C++
enum optAssertionEdge
{
    OAE_FLOW_INSENSITIVE,
    OAE_NEXT_EDGE,             // assertion holds for bbNext
    OAE_JUMP_EDGE              // assertion holds for bbJumpDest
};
```
Also, it may be better to not use op2 to store this information, there already seems to be some weirdness in the way assertion operands and their `vn` members are handled.

Anyway, I'm still looking for alternatives in this area. The obvious solution would be to add the complementary `OAK_THROW` like all other JTRUE generated assertions do. But such an assertion is unlikely to provide any useful information yet burn an assertion table entry and we don't have too many entries to begin with.